### PR TITLE
[1245] added ability to configure customizer at global level

### PR DESCRIPTION
**Pull request by Manikandan-K**
This fix is related to issue https://github.com/jdbi/jdbi/issues/1245

### CODE REVIEW

**Comment by 28Smiles** [comment]
Looks good to me, but I would suggest a mention in the docu of this feature.

File: core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java [hunk 1]
```diff
@@ -50,6 +54,7 @@ private SqlStatements(SqlStatements that) {
         this.sqlLogger = that.sqlLogger;
         this.queryTimeout = that.queryTimeout;
         this.allowUnusedBindings = that.allowUnusedBindings;
+        this.customizers = that.customizers;
```
Comment (TheRealMarnes) [inline core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java]: I think this should be a copy, not a reference. We wouldn't want changes to one jdbi context changing the behavior of another.

File: core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java [hunk 2]
```diff
@@ -34,13 +36,15 @@
...
     private boolean allowUnusedBindings;
+    private final Collection<StatementCustomizer> customizers;
 
     public SqlStatements() {
         attributes = new ConcurrentHashMap<>();
         templateEngine = new DefinedAttributeTemplateEngine();
         sqlParser = new ColonPrefixSqlParser();
         sqlLogger = SqlLogger.NOP_SQL_LOGGER;
         queryTimeout = null;
+        customizers = new ArrayList<>();
```
Comment (TheRealMarnes) [inline core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java]: Shouldn't this maybe be a threadsafe list?

**Comment by TheRealMarnes** [comment]
@28Smiles when you're entirely satisfied with the changes (just waiting for documentation?), I guess I'll go ahead and merge.

**Comment by 28Smiles** [comment]
@TheRealMarnes yes, just waiting for the doc

**Comment by TheRealMarnes** [comment]
Looks good to me then. Thanks @Manikandan-K

**Comment by TheRealMarnes** [comment]
Looks good to me then. Thanks @Manikandan-K

**Pull request merged**